packages/asset-swapper/src/utils/market_operation_utils/comparison_price.ts

```diff
@@ -4,56 +4,42 @@ import * as _ from 'lodash';
 
 import { MarketOperation } from '../../types';
 
-import { COMPARISON_PRICE_DECIMALS, SOURCE_FLAGS } from './constants';
-import { ComparisonPrice, MarketSideLiquidity, OptimizerResult } from './types';
+import { COMPARISON_PRICE_DECIMALS } from './constants';
+import { ComparisonPrice, ERC20BridgeSource, FeeEstimate, FeeSchedule, MarketSideLiquidity } from './types';
 
 /**
  * Takes in an optimizer response and returns a price for RFQT MMs to beat
  * returns the price of the taker asset in terms of the maker asset
  * So the RFQT MM should aim for a higher price
- * @param optimizerResult the output of the optimizer, contains the best orders
+ * @param adjustedRate the adjusted rate (accounting for fees) from the optimizer, maker/taker
  * @param amount the amount specified by the client
  * @param marketSideLiquidity the results from querying liquidity sources
+ * @param feeSchedule the fee schedule passed to the Optimizer
  * @return ComparisonPrice object with the prices for RFQ MMs to beat
  */
 export function getComparisonPrices(
-    optimizerResult: OptimizerResult,
+    adjustedRate: BigNumber,
     amount: BigNumber,
     marketSideLiquidity: MarketSideLiquidity,
+    feeSchedule: FeeSchedule,
 ): ComparisonPrice {
-    // the adjusted rate is defined as maker/taker
-    // input is the taker token for sells, input is the maker token for buys
-    const makerTakerOptimalRate = optimizerResult.adjustedRate;
-    const takerMakerOptimalRate = optimizerResult.adjustedRate.pow(-1);
-
-    // fees for a native order
-    const fees = optimizerResult.exchangeProxyOverhead(SOURCE_FLAGS.Native);
-
     // Calc native order fee penalty in output unit (maker units for sells, taker unit for buys)
     const feePenalty = !marketSideLiquidity.ethToOutputRate.isZero()
-        ? marketSideLiquidity.ethToOutputRate.times(fees)
+        ? marketSideLiquidity.ethToOutputRate.times((feeSchedule[ERC20BridgeSource.Native] as FeeEstimate)(undefined))
```

Comment:
I'm not sure I get what you are doing here, why is there `undefined` here?

Reply:
We should prob comment this with a `// HACK:`.

Reply:
Might as well put it into a variable as well since it appears twice.

Reply:
Talked to @merklejerk about this. It takes in a `Fill` object, but we don't have one here of course. This should return the cost of filling a single native order, and from testing it looks like that's the case.

I'm going to test one more time after making final edits.

I'll put a comment explaining the intended use.

Reply:
It works bc the native order schedule just returns a constant value: https://github.com/0xProject/protocol/blob/4f82543bdf361dc82d5fa8eaf728550033003907/packages/asset-swapper/src/utils/market_operation_utils/constants.ts#L441

If the gas schedule for native orders had a dependency on the `fillData` tho, I think it wouldn't work.